packages/cozy-mespapiers-lib/src/components/OnboardedGuardedRoute.jsx

```diff
@@ -1,50 +1,50 @@
 import React from 'react'
-import { Route, Redirect } from 'react-router-dom'
+import { Navigate, Outlet, useLocation } from 'react-router-dom'
 
 import { useQuery, hasQueryBeenLoaded } from 'cozy-client'
 import Spinner from 'cozy-ui/transpiled/react/Spinner'
 
 import { getOnboardingStatus } from '../helpers/queries'
+
 import { useOnboarding } from './Hooks/useOnboarding'
 
-const OnboardedGuardedRoute = ({ component: Component, render, ...rest }) => {
+const OnboardedGuardedRoute = () => {
+  const location = useLocation()
+  const { OnboardingComponent } = useOnboarding()
+  const isOnboardingPage = location.pathname === '/paper/onboarding'
+
   const { data: settingsData, ...settingsQuery } = useQuery(
     getOnboardingStatus.definition,
     getOnboardingStatus.options
   )
+
+  if (!hasQueryBeenLoaded(settingsQuery)) {
+    return (
+      <Spinner
+        size="xxlarge"
+        className="u-flex u-flex-justify-center u-mt-2 u-h-5"
+      />
+    )
+  }
+
   const onboarded = settingsData?.[0]?.onboarded
-  const { OnboardingComponent } = useOnboarding()
 
-  return !hasQueryBeenLoaded(settingsQuery) ? (
-    <Spinner
-      size="xxlarge"
-      className="u-flex u-flex-justify-center u-mt-2 u-h-5"
-    />
-  ) : (
-    <Route
-      {...rest}
-      render={props => {
-        const isOnboardingPage = rest?.path === '/paper/onboarding'
-
-        if (
-          (isOnboardingPage && onboarded === true) ||
-          (isOnboardingPage && !OnboardingComponent)
-        ) {
-          return <Redirect to="/paper" />
-        } else if (
-          !isOnboardingPage &&
-          onboarded !== true &&
-          OnboardingComponent
-        ) {
-          return <Redirect to="/paper/onboarding" />
-        } else if (Component) {
-          return <Component {...props} />
-        } else {
-          return render(props)
-        }
-      }}
-    />
-  )
+  const isAlreadyOnboarded =
+    (isOnboardingPage && onboarded === true) ||
+    (isOnboardingPage && !OnboardingComponent)
+
+  const isNotOnboarded =
+    !isOnboardingPage && onboarded !== true && OnboardingComponent
+
+  if (isAlreadyOnboarded) {
+    return <Navigate to="/" replace />
+  }
+
+  if (isNotOnboarded) {
+    return <Navigate to="onboarding" replace />
+  }
+
+  return <Outlet />
```

Comment:
nit: qu'est-ce que fait Outlet ?

Reply:
https://reactrouter.com/en/v6.3.0/api#outlet 😗

Reply:
arf j'étais preneur d'un petit blabla, ça fait toujours plaisir... je suis tristesse

Reply:
> *An <Outlet> should be used in parent route elements to render their child route elements. This allows nested UI to show up when child routes are rendered. If the parent route matched exactly, it will render a child index route or nothing if there is no index route.*

Comme ça ? :trollface: